packages/core/src/guided-answers-api.ts

```diff
@@ -209,72 +208,40 @@ function isHtmlExtensionApplicable(ide: IDE, extension: GuidedAnswerHtmlExtensio
-    nodeExtensions: GuidedAnswerNodeExtension[] = []
+    enhancements: GuidedAnswersEnhancement[] = []
 ): { htmlEnhancements: HTMLEnhancement[]; nodeCommands: Command[] } {
     const nodeCommands: Command[] = [];
     const htmlEnhancements: HTMLEnhancement[] = [];
     try {
-        const applicableNodeExtensions = nodeExtensions.filter((nodeExt) =>
-            isExtensionApplicable(ide, nodeExt, extensions)
-        );
-        for (const applicableNodeExtension of applicableNodeExtensions) {
-            const label = applicableNodeExtension.LABEL;
-            const description = applicableNodeExtension.DESCRIPTION;
-            const exec =
-                applicableNodeExtension.TYPE === 'Extension Command'
-                    ? { extensionId: applicableNodeExtension.ARG1.VALUE, commandId: applicableNodeExtension.ARG2.VALUE }
-                    : {
-                          cwd: applicableNodeExtension.ARG1.VALUE,
-                          arguments: applicableNodeExtension.ARG2.VALUE.split(' ')
-                      };
-            nodeCommands.push({ label, description, exec });
-        }
-
-        const applicableHtmlExtensions = htmlExtensions.filter((htmlExt) =>
-            isHtmlExtensionApplicable(ide, htmlExt, extensions)
-        );
-        for (const applicableHtmlExtension of applicableHtmlExtensions) {
-            const text = applicableHtmlExtension.text;
-            const label = applicableHtmlExtension.label;
-            const description = applicableHtmlExtension.desc;
-            let exec: TerminalCommand | VSCodeCommand;
-            if (applicableHtmlExtension.command.type === 'Extension') {
-                const extensionId = applicableHtmlExtension.command.exec.extensionId;
-                const commandId = applicableHtmlExtension.command.exec.command;
-                let argument;
-                try {
-                    argument = applicableHtmlExtension.command.exec.args
-                        ? JSON.parse(applicableHtmlExtension.command.exec.args)
-                        : undefined;
-                } catch (error) {
-                    logger.logString(
-                        `Error when parsing argument '${
-                            applicableHtmlExtension.command.exec.args
-                        }' for HTML enhancement. Extension id: '${extensionId}', command id: '${commandId}', ${error?.toString()}`
-                    );
-                }
-                exec = { extensionId, commandId, argument };
-            } else {
-                exec = {
-                    arguments: [
-                        applicableHtmlExtension.command.exec.command,
-                        ...(applicableHtmlExtension.command.exec.args ?? '').split(' ')
-                    ]
-                };
+        for (const enhancement of enhancements) {
+            if (
+                (ide === 'VSCODE' && enhancement.command.environment.vscode !== 1) ||
```

Comment:
Extracting/deconstructing `enhancement.command.environment` before the loop can enhance readability if used in multiple places

Reply:
I could do
```typescript
const { sbas, vscode } = enhancement.command.environment;
if (
    (ide === 'VSCODE' && vscode !== 1) ||
    (ide === 'SBAS' && sbas !== 1) ||
    (enhancement.command.type === 'Extension' && !extensions.has(enhancement.command.exec.context))
) {
    continue;
}
```

but as you've wrote, this would make sense if `sbsas` and `vscode` is used in other places. As it is used only once I think it is fine to refer to it directly without destructing beforehand.